### ISSUE TITLE
How to keep caps?

### ISSUE THREAD

**Issue by unhammer** [apertium/apertium-separable#36]
```
 $ echo '^Den<det><dem><nt><sg>$ ^enkelt<adj><pst><un><sp><def>$ ^departement<n><nt><sg><ind>$'|lsx-proc nob-nno.autoseq.bin
^hver<det><qnt><nt><sg>$ ^enkelt<adj><pst><nt><sg><ind>$ ^departement<n><nt><sg><ind>$
 
– how do you get it to carry over the caps?

lsx has `  <p><l>den<s n="det"/></l><r>hver<s n="det"/></r></p>`

**Comment by mr-martian** [comment]
Currently the code always outputs dictionary case.

The way other tools handle this is to check the case of the input word and set the case of the output to be that as well, but that presents a problem since the different words can have different capitalizations (as in your example).

We would be closer to solving this if we implemented a mechanism of specifying which words corresponded in the input and output, though I think the there would still need to be a decision about what would be the "correct" solution.

See also https://github.com/apertium/apertium/issues/67

**Comment by unhammer** [comment]
The way biltrans `lt-proc -b` does things is that if dictionary is upper case, require upper case, while if it's lower case we allow upper and if so keep it. It's not perfect (you can't force-downcase things), but it easily solves 95% of stuff (beginning-of-sentence capitalisation). Since separable goes right before biltrans and does basically the same just spanning `^ $`, it'd be nice if it did the same so we don't have to change the whole pipeline to use it.

**Comment by mr-martian** [comment]
Ok, so if the input is `Aa aa aa`, `aa aa aa`, or `AA AA AA`, it's relatively clear what to do, but what should happen if the case varies in other ways? Or should it be the normal thing of tracking case by looking at first letter and last letter but across the whole phrase instead of a single LU?

**Comment by unhammer** [comment]
The existing pipeline can't do anything more complicated than aa/Aa/AA by looking at the first/last letter; any heuristic here shouldn't try for anything more complicated. I can see how uppercasing whole phrases using just first/last letter might be dangerous though, so if only titlecasing happens that would be fine by me.

It should probably be possible to switch off the heuristic. (If we had "real" case handling, the input itself would be lemma-case all the way from morph analysis through to generation and no tools should change case until generation.)

**Comment by mr-martian** [comment]
Yeah, I was just overcomplicating it in my head. I'll do first letter of first word and last letter of last word and include an equivalent to `lt-proc`'s `-w`.

**Comment by mr-martian** [comment]
Implemented in 3a02116
